### PR TITLE
Removed renderForm and fixed typings for InputBase

### PR DESCRIPTION
**Pull request by m1aw**
## Summary

This PR tries to fix the main problem with using some of the Form inputs, which was loss of Typescript type safety, since it there where no property checks in `renderForm`.
The approach taken was to fix InputBase typings so from the Component all the way to the input there was type safety. There was a need for some minor clean up and touches in some of the components. Mainly components there were infringing on the types set. 
Notably Checkbox which was reaving a "value" that was a boolean, while the "booelan" was read from the `checked` attribute.
In the end the it was also necessary to finally remove `renderFormField` because it couldn't guarantee which type of component was being used, which was once again breaking the types. 

## Tests

All the unit tests are passing, tho some should be improved in the future. 

**Fixed issue**:

### CODE REVIEW

File: packages/lib/src/components/internal/StoreDetails/StoreDetails.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import useCoreContext from '../../../core/Context/useCoreContext';
 /**
  * "Store details" generic checkbox
  */
-function StoreDetails({ storeDetails = false, ...props }) {
+function StoreDetails({ storeDetails = true, ...props }) {
```
Comment (sponglord) [inline packages/lib/src/components/internal/StoreDetails/StoreDetails.tsx:9]: why the change from `false` to `true`?

File: packages/lib/src/components/BacsDD/components/BacsInput.tsx [hunk 2]
```diff
@@ -86,20 +87,20 @@ function BacsInput(props: BacsInputProps) {
...
-                    onBlur: handleChangeFor('holderName', 'blur'),
-                    onInput: handleChangeFor('holderName', 'input')
-                })}
+                <InputText
+                    name={'bacs.accountHolderName'}
+                    className={'adyen-checkout__bacs-input--holder-name'}
+                    placeholder={props.placeholders.holderName}
+                    value={data.holderName}
+                    aria-invalid={!valid.holderName}
+                    aria-label={i18n.get('bacs.accountHolderName')}
```
[the remark this answers is not in the document]
Comment (m1aw) [reply packages/lib/src/components/BacsDD/components/BacsInput.tsx:96]: We can. It doesn't matter for Preact, so I kept the same format we had before. Made it simpler to convert and check for diffs.

File: packages/lib/src/components/internal/Address/components/FieldContainer.tsx [hunk 3]
```diff
@@ -67,17 +67,19 @@ function FieldContainer(props: FieldContainerProps) {
...
+                        name={fieldName}
+                        {...{
+                            classNameModifiers,
+                            value,
+                            onInput,
+                            onBlur,
+                            maxLength,
+                            trimOnBlur,
+                            disabled
+                        }}
```
[the remark this answers is not in the document]
Comment (m1aw) [reply packages/lib/src/components/internal/Address/components/FieldContainer.tsx:80]: I don't think that is valid JSX.
I honestly just made it like that because it was simpler to convert from on to another.

File: packages/lib/src/components/internal/StoreDetails/StoreDetails.test.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    const checkbox = await screen.findByRole('checkbox');
+    expect(checkbox).not.toBeChecked();
+
+    await user.click(checkbox);
+    expect(checkbox).toBeChecked();
+});
+
+test('StoredDetails storeDetails prop true does nothing LEGACY TEST', async () => {
+    // I wanted to capture this buggy feature,
+    // it should be removed but I wanted to test to make sure there where no side effects
```
[the remark this answers is not in the document]
Comment (m1aw) [reply packages/lib/src/components/internal/StoreDetails/StoreDetails.test.tsx:20]: I'd say it doesn't hurt to keep for documentation reasons.

[the remark this answers is not in the document]
Comment (ribeiroguilherme) [reply packages/lib/src/components/internal/StoreDetails/StoreDetails.test.tsx:20]: I meant the comment, sorry

File: packages/lib/src/components/internal/Address/components/FieldContainer.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (ribeiroguilherme) [reply packages/lib/src/components/internal/Address/components/FieldContainer.tsx:80]: Indeed it is not valid . It is valid if we use boolean values, but not using objects/functions

**Comment by sponglord** [comment]
So can the `FormFields/index.ts` be removed?

File: packages/lib/src/components/internal/Address/components/FieldContainer.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (sponglord) [reply packages/lib/src/components/internal/Address/components/FieldContainer.tsx:80]: It is weird that this is done differently to elsewhere.
It should just be:
```
<InputText
                name={fieldName}
                classNameModifiers={classNameModifiers}
                value={value}
               ... etc
            />
```

File: packages/lib/src/components/internal/FormFields/InputBase.tsx [hunk 5]
```diff
@@ -1,17 +1,33 @@
...
+    isValid?: boolean;
+    readonly?: boolean;
+    uniqueId?: string;
+    disabled?: boolean;
+    className?: string;
+    placeholder?: string;
+    value?: string;
+    name?: string;
+    checked?: boolean;
+    setRef?: (ref: MutableRef<EventTarget>) => void;
```
Comment (sponglord) [inline packages/lib/src/components/internal/FormFields/InputBase.tsx:19]: How/Where is `setRef` used?

File: packages/lib/src/components/internal/StoreDetails/StoreDetails.tsx [hunk 6]
```diff
@@ -20,12 +20,7 @@ function StoreDetails({ storeDetails = false, ...props }) {
 
     return (
         <div className="adyen-checkout__store-details">
-            {renderFormField('boolean', {
-                onChange,
-                label: i18n.get('storeDetails'),
-                value,
-                name: 'storeDetails'
-            })}
+            <Checkbox onChange={onChange} label={i18n.get('storeDetails')} name={'storeDetails'} />
```
Comment (sponglord) [inline packages/lib/src/components/internal/StoreDetails/StoreDetails.tsx:23]: Shouldn't the previous `value` property be passed as: `checked: !!value`?
Doesn't it serve as a way to say whether the checkbox comes "pre-checked"?

File: packages/lib/src/components/internal/SendCopyToEmail/SendCopyToEmail.tsx [hunk 7]
```diff
@@ -18,12 +19,7 @@ export default function SendCopyToEmail(props) {
     return (
         <div className={cx('adyen-checkout__fieldset', 'adyen-checkout__fieldset--sendCopyToEmail', props.classNames)}>
             <Field classNameModifiers={['sendCopyToEmail']} name={'sendCopyToEmail'} useLabelElement={false} addContextualElement={false}>
-                {renderFormField('boolean', {
-                    onChange: toggleEmailField,
-                    label: i18n.get('boleto.sendCopyToEmail'),
-                    name: 'sendCopyToEmail',
-                    value: sendCopyToEmail
-                })}
+                <Checkbox onChange={toggleEmailField} label={i18n.get('boleto.sendCopyToEmail')} name={'sendCopyToEmail'}/>
```
Comment (sponglord) [inline packages/lib/src/components/internal/SendCopyToEmail/SendCopyToEmail.tsx:22]: Shouldn't the previous `value` property be passed as: `checked: !!value`?
Doesn't it serve as a way to say whether the checkbox comes "pre-checked"?
